### DISCUSSION TITLE
Grouped country data

### DISCUSSION THREAD

**Discussion by dil-abertok** [bchavez/Bogus#524] [Ideas]
I just run into the issue of not being able to generate country data (e.g. country name and country code) that belongs together, as opposed to e.g. the `Person()` method, or the `Currency`, that contains the currency name and the currency code.
The full dataset I am looking for is something like this:
```csharp
var country = faker.Country();
country.Dump();

/*
{
  "Name" : "United States"
  "CountryCode" : {
    "Alpha2" : "US",
    "Alpha2" : "USA"
  },
  "Currency" : {
    "Code" : "USD",
    "Description" : "US Dollar",
    "Symbol" : "$"
  }
}
*/
```
I have the feeling that this is a limitation of the faker dataset that is not structured in a way that associating this data is trivial. Am I correct to assume that this would probably be a dataset change request in Faker.js?

**Comment by bchavez** [comment]
Hi. Yes, that's correct.

The country code is just a list/array of string:
* https://github.com/bchavez/Bogus/blob/9d6c22cd6b75c2d7ddd36c0960665eb8171f5b37/Source/Bogus/data/en.locale.json#L289-L294

And currency is slightly more structured, but not exactly in the way you're looking for:
* https://github.com/bchavez/Bogus/blob/9d6c22cd6b75c2d7ddd36c0960665eb8171f5b37/Source/Bogus/data/en.locale.json#L8157-L8167

The easiest is to find a CSV or some kind of list/data that has the object structure you're looking for and throw an extension method around your custom data & if you want to use Bogus' `.Random` APIs.
